bpf/logenricher/types.h

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <bpfcore/vmlinux.h>
+#include <bpfcore/bpf_core_read.h>
+#include <bpfcore/bpf_helpers.h>
+
+#include <common/tp_info.h>
+
+enum {
+    // include/linux/tty_driver.h
+    k_tty_driver_type_pty = 0x0004,
+    k_tty_driver_subtype_pty_master = 0x0001,
+
+    // include/uapi/asm-generic/termbits.h
+    k_echo = 0x00008,
+
+    // log handling
+    k_log_event_max_size = 1 << 14,                       // 16K
+    k_log_event_max_size_mask = k_log_event_max_size - 1, // 16K - 1
+    k_log_event_max_log_mask = (1 << 13) - 1,             // 8K - 1
+
+    // terminal file
+    k_pts_file_path_len_max = 64,
+    k_pts_file_path_len_max_mask = k_pts_file_path_len_max - 1,
+};
+
+typedef struct log_event {
+    tp_info_pid_t pid_tp;
+    u32 tgid;
+    u32 len;
+    u32 fd;
+    u32 _pad;
+    u8 file_path[k_pts_file_path_len_max];
+    u8 log[];
+} log_event_t;
+
+const log_event_t *log_event__unused __attribute__((unused));
+
+enum tty_driver_type__new {
+    TTY_DRIVER_TYPE_SYSTEM,
+    TTY_DRIVER_TYPE_CONSOLE,
+    TTY_DRIVER_TYPE_SERIAL,
+    TTY_DRIVER_TYPE_PTY,
+    TTY_DRIVER_TYPE_SCC,
+    TTY_DRIVER_TYPE_SYSCONS,
+};
+
+enum tty_driver_subtype__new {
```

Comment:
```suggestion
enum tty_driver_subtype___new {
```

the mechanism requires three underscores - I think we got lucky here.

Ref: https://kernel.googlesource.com/pub/scm/linux/kernel/git/torvalds/linux/%2B/4333a9b0b67bb4e8bcd91bdd80da80b0ec151162/tools/lib/bpf/libbpf.c#4019